packages/ra-ui-materialui/src/input/AutocompleteInput.tsx

```diff
@@ -213,6 +214,13 @@ export const AutocompleteInput = <
         ...rest,
     });
 
+    const finalOnBlur = useCallback((): void => {
+        if (clearOnBlur && !(!!create || !!onCreate)) {
+            setFilterValue('');
+        }
```

Comment:
I don't understand, why clear the filter only if you don't have `create` or `onCreate` props?

Reply:
Is our custom `freeSolo` condition ( `freeSolo={!!create || !!onCreate}`).
At first I was asking `if (clearOnBlur && !props.freeSolo) {`

Reply:
Not sure I understand :grimacing: 
According to https://mui.com/material-ui/react-autocomplete/#creatable , we should have `clearOnBlur=true` when `freeSolo=true`.

Since having `clearOnBlur=true` on MUI's `<Autocomplete>` does not seem to work, to me the condition should be `if (clearOnBlur || props.freeSolo)` rather than `if (clearOnBlur && !props.freeSolo)`
Or simply `if (clearOnBlur)` since we encourage `clearOnBlur=true` by default anyway.

Again, not sure I understand your point completely, but I tested your branch with the simple example, and it feels weird to still have something inside the Autocomplete when I click away and did not create a new item.

TLDR; I'd simply go with `if (clearOnBlur)`